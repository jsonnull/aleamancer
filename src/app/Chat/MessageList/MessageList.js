/* @flow */
import React from 'react'
import MessageView from '../Message'
import type { Message } from 'types'
import styles from './style.css'

type Props = {
  messages: Array<Message>
}

export default class MessageList extends React.Component<*, Props, *> {
  timer: number
  scroll: HTMLElement

  componentDidMount () {
    // Every minute, update chat timestamps
    this.timer = setInterval(() => this.forceUpdate(), 60000)
  }

  componentDidUpdate () {
    this.scroll.scrollTop += 10000
  }

  componentWillUnmount () {
    clearInterval(this.timer)
  }

  render () {
    return <div
      className={ styles.messages }
      ref={el => { this.scroll = el }}
    >
      {this.props.messages.map(message =>
        <MessageView key={message.key} message={message} />
      )}
      <div className={ styles.spacer }>&nbsp;</div>
    </div>
  }
}
