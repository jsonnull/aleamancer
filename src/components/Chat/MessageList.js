// @flow
import * as React from 'react'
import styled from 'styled-components'
import MessageView from './Message'
import type { Message } from 'types'

const MessagesWrapper = styled.div`
  min-height: 0;
  flex: 1;
`
const Messages = styled.div`
  overflow-y: ${props => (props.isPinned ? 'scroll' : 'hidden')};
  overflow-x: hidden;
  width: 315px;
  height: 100%;

  display: flex;
  flex-direction: column;
  flex: ${props => (props.isPinned ? '1' : 'none')};

  align-items: ${props => (props.isPinned ? '' : 'flex-end')};
`

type Props = {
  messages: Array<Message>,
  isPinned: boolean
}

export default class MessageList extends React.Component<Props> {
  timer: number
  scroll: ?HTMLElement

  componentDidMount() {
    // Every minute, update chat timestamps
    this.timer = setInterval(() => this.forceUpdate(), 60000)
  }

  componentDidUpdate() {
    if (this.scroll) {
      this.scroll.scrollTop += 10000
    }
  }

  componentWillUnmount() {
    clearInterval(this.timer)
  }

  render() {
    const { isPinned } = this.props
    return (
      <MessagesWrapper>
        <Messages
          isPinned={isPinned}
          innerRef={el => {
            this.scroll = el
          }}
        >
          {this.props.messages.map(message => (
            <MessageView
              key={message.id}
              message={message}
              isPinned={isPinned}
            />
          ))}
        </Messages>
      </MessagesWrapper>
    )
  }
}
