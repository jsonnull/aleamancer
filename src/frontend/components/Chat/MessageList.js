// @flow
import * as React from 'react'
import styled from 'styled-components'
import type { Message } from 'common/types'
import MessageView from './Message'

const MessagesWrapper = styled.div`
  min-height: 0;
  flex: 1;
`
const Messages = styled.div`
  overflow-y: ${props => (props.isPinned ? 'scroll' : 'hidden')};
  overflow-x: hidden;
  width: 315px;
  height: 100%;

  display: flex;
  flex-direction: column;
  flex: ${props => (props.isPinned ? '1' : 'none')};

  align-items: ${props => (props.isPinned ? '' : 'flex-end')};
`

type Props = {
  messages: Array<Message>,
  isPinned: boolean
}

export default class MessageList extends React.Component<Props> {
  timer: IntervalID
  scroll: ?HTMLElement

  componentDidMount() {
    // Every minute, update chat timestamps
    this.timer = setInterval(() => this.forceUpdate(), 60000)
  }

  componentDidUpdate() {
    if (this.scroll) {
      this.scroll.scrollTop += 10000
    }
  }

  componentWillUnmount() {
    clearInterval(this.timer)
  }

  render() {
    const { isPinned } = this.props
    return (
      <MessagesWrapper>
        <Messages
          isPinned={isPinned}
          innerRef={el => {
            this.scroll = el
          }}
        >
          {this.props.messages.map(message => (
            <MessageView
              key={message.id}
              message={message}
              isPinned={isPinned}
            />
          ))}
        </Messages>
      </MessagesWrapper>
    )
  }
}
