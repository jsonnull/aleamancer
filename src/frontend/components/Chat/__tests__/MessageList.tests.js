// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import MessageList from '../MessageList.js'

describe('Chat MessageList component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MessageList messages={[]} isPinned={false} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
