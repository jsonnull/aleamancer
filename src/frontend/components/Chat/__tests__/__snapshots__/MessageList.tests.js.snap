// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat MessageList component renders correctly 1`] = `
.c0 {
  min-height: 0;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 {
  overflow-y: hidden;
  overflow-x: hidden;
  width: 315px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
